packages/@uppy/status-bar/src/StatusBar.js

```diff
@@ -64,68 +39,131 @@ module.exports = (props) => {
     hideCancelButton,
     hideRetryButton,
     recoveredState,
+    uploadState,
+    totalProgress,
+    files,
+    supportsUploadProgress,
+    hideAfterFinish,
+    isSomeGhost,
+    isTargetDOMEl,
+    doneButtonHandler,
+    isUploadStarted,
+    i18n,
+    startUpload,
+    uppy,
+    isAllComplete,
+    showProgressDetails,
+    numUploads,
+    complete,
+    totalSize,
+    totalETA,
+    totalUploadedSize,
   } = props
 
-  const { uploadState } = props
+  function getProgressValue () {
+    switch (uploadState) {
+      case STATE_POSTPROCESSING:
+      case STATE_PREPROCESSING: {
+        const progress = calculateProcessingProgress(files)
 
-  let progressValue = props.totalProgress
-  let progressMode
-  let progressBarContent
+        if (progress.mode === 'determinate') {
+          return progress.value * 100
+        }
+        return totalProgress
+      }
+      case STATE_ERROR: {
+        return null
+      }
+      case STATE_UPLOADING: {
+        if (!supportsUploadProgress) {
+          return null
+        }
+        return totalProgress
+      }
+      default:
+        return totalProgress
+    }
+  }
 
-  if (uploadState === statusBarStates.STATE_PREPROCESSING || uploadState === statusBarStates.STATE_POSTPROCESSING) {
-    const progress = calculateProcessingProgress(props.files)
-    progressMode = progress.mode
-    if (progressMode === 'determinate') {
-      progressValue = progress.value * 100
+  function getIsIndeterminate () {
+    switch (uploadState) {
+      case STATE_POSTPROCESSING:
+      case STATE_PREPROCESSING: {
+        const { mode } = calculateProcessingProgress(files)
+        return mode === 'indeterminate'
+      }
+      case STATE_UPLOADING: {
+        if (!supportsUploadProgress) {
+          return true
+        }
+        return false
+      }
+      default:
+        return false
     }
+  }
 
-    progressBarContent = ProgressBarProcessing(progress)
-  } else if (uploadState === statusBarStates.STATE_COMPLETE) {
-    progressBarContent = ProgressBarComplete(props)
-  } else if (uploadState === statusBarStates.STATE_UPLOADING) {
-    if (!props.supportsUploadProgress) {
-      progressMode = 'indeterminate'
-      progressValue = null
+  function getIsHidden () {
+    if (recoveredState) {
+      return false
     }
 
-    progressBarContent = ProgressBarUploading(props)
-  } else if (uploadState === statusBarStates.STATE_ERROR) {
-    progressValue = undefined
-    progressBarContent = ProgressBarError(props)
+    switch (uploadState) {
+      case STATE_WAITING:
+        if (hideUploadButton) {
+          return true
+        }
+        if (!newFiles > 0) {
```

Comment:
Is the `!` a typo? Currently that's equivalent to `newFiles !== 0`, which is fine as `newFiles` should never be negative, but that's terrible in terms of readability 😅

Reply:
I just copy pasted this here. You are right.  Will change it.